Islandora-CLAW/Alpaca
# Triplestsore indexer

https://github.com/Islandora-CLAW/CLAW/issues/394

This adds a small camel connector to index RDF from CLAW into a triplestore.

## To Test
Assuming you're starting from scratch here (it's easier than getting lost in Karaf)
* Pull in https://github.com/Islandora-CLAW/claw_vagrant/pull/8 into your claw_vagrant repo.
  * `$ cd /path/to/claw_vagrant`
  * `$ git pull https://github.com/dannylamb/claw_vagrant triplestore-indexer`
* Update `alpaca.sh` to say `git clone -b triplestsore-indexer https://github.com/dannylamb/Alpaca.git` instead of cloning from origin/master.  Be mindful of the spelling on that branch name.  I derped pushing it up to Github
* `vagrant up`
* When it's finished, uninstall the `islandora` module (you will have to disable `islandora_collection` too)
* Log into the box with `vagrant ssh`
* Pull in the changes from https://github.com/Islandora-CLAW/islandora/pull/34, which will enable GET requests on FedoraResource entities.
  * `$ cd /var/www/html/drupal/web/modules/contrib/islandora`
  * `$ git pull https://github.com/dannylamb/islandora.git triplestore-indexer`
* Then re-install `islandora` and `islandora_collection` modules.

Once installed:
* Visit `http://localhost:8000/admin/content/fedora_resource` and create a new Fedora resource
![screenshot from 2017-02-01 14-50-42](https://cloud.githubusercontent.com/assets/20773151/22521554/c367ad5e-e88e-11e6-932b-622f71236cf8.png)
* Visit `http://localhost:8080/bigdata/#query` and issue a `select ?s ?p ?o where { ?s ?p ?o }` query to get all triples.  You should see the triples from the rdf mapping for the bundle you created.
![screenshot from 2017-02-01 14-51-02](https://cloud.githubusercontent.com/assets/20773151/22521971/3961d650-e890-11e6-92bf-2b01e706ad12.png)
* Visit your resource in Drupal and update it
![screenshot from 2017-02-01 14-51-37](https://cloud.githubusercontent.com/assets/20773151/22521998/523f7704-e890-11e6-9f6c-aef05049faf1.png)
* Go back to the triplestore and run the query again.  At a minimum, modified dates and vclock should have updated.
![screenshot from 2017-02-01 14-51-53](https://cloud.githubusercontent.com/assets/20773151/22522018/679cb256-e890-11e6-9b3b-bfc0d6588fa7.png)
* Visit your resource in Drupal again, and delete it.
![screenshot from 2017-02-01 14-52-12](https://cloud.githubusercontent.com/assets/20773151/22522036/746d5760-e890-11e6-8aa6-a6bd2c00d8f0.png)
* Back in the triplestore, the triples should be gone and the query will return no results.
![screenshot from 2017-02-01 14-52-30](https://cloud.githubusercontent.com/assets/20773151/22522056/885bae3e-e890-11e6-9853-2811ab504561.png)
